File: Carnicus/Assets/NetworkModels/Updates/KeypressUpdate.cs

using UnityEngine;
using Newtonsoft.Json;

public class KeypressUpdate : ModelInterface
{
	public string type = "KeypressUpdate";
	public bool up;
	public bool down;
	public bool left;
	public bool right;
	public bool shift;
	public bool attack;
	public bool attack2;
	public bool attack3;
	public float diff;
	public int inputSequenceNumber;
	public float forwardX;
	public float forwardY;
	public float forwardZ;

	public bool noInput;
	
	public KeypressUpdate(bool up, bool down, bool left, bool right, bool shift, bool attack, bool attack2, bool attack3, float diff, int inputSequenceNumber, 
	                      float forwardX, float forwardY, float forwardZ, bool noInput)
	{
		this.up = up;
		this.down = down;
		this.left = left;
		this.right = right;
		this.shift = shift;
		this.attack = attack;
		this.attack2 = attack2;
		this.attack3 = attack3;
		this.diff = diff;
		this.inputSequenceNumber = inputSequenceNumber;
		this.forwardX = forwardX;
		this.forwardY = forwardY;
		this.forwardZ = forwardZ;
		this.noInput = noInput;
	}
	
	public override string ToNetworkString()
	{
		return JsonConvert.SerializeObject(this);
	}
	
	public static KeypressUpdate ToModel(string networkString)
	{
		KeypressUpdate req = JsonConvert.DeserializeObject<KeypressUpdate>(networkString);
		if (req != null && req.type == "KeypressUpdate")
			return req;
		return null;
	}
}

File: CarnicusServer/src/main/java/com/malow/carnicusserver/models/updates/KeypressUpdate.java
package com.malow.carnicusserver.models.updates;

import java.io.IOException;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.malow.carnicusserver.models.ModelInterface;


public class KeypressUpdate implements ModelInterface
{
    public final boolean up;
    public final boolean down;
    public final boolean left;
    public final boolean right;
    public final boolean shift;
    public final boolean attack;
    public final boolean attack2;
    public final boolean attack3;
    public final float diff;
	public final int inputSequenceNumber;
	public final float forwardX;
	public final float forwardY;
	public final float forwardZ;
    public final boolean noInput;

    @JsonCreator
    public KeypressUpdate(@JsonProperty("up") boolean up,
    				      @JsonProperty("down") boolean down, 
    				      @JsonProperty("left") boolean left, 
    				      @JsonProperty("right") boolean right, 
    				      @JsonProperty("shift") boolean shift, 
    				      @JsonProperty("attack") boolean attack, 
    				      @JsonProperty("attack2") boolean attack2, 
    				      @JsonProperty("attack3") boolean attack3, 
    				      @JsonProperty("diff") float diff,
    				      @JsonProperty("inputSequenceNumber") int inputSequenceNumber,
    				      @JsonProperty("forwardX") float forwardX,
    				      @JsonProperty("forwardY") float forwardY,
    				      @JsonProperty("forwardZ") float forwardZ, 
    				      @JsonProperty("noInput") boolean noInput)
    {
        this.up = up;
        this.down = down;
        this.left = left;
        this.right = right;
        this.shift = shift;
        this.attack = attack;
        this.attack2 = attack2;
        this.attack3 = attack3;
        this.diff = diff;
    	this.inputSequenceNumber = inputSequenceNumber;
    	this.forwardX = forwardX;
    	this.forwardY = forwardY;
    	this.forwardZ = forwardZ;
    	this.noInput = noInput;
    }

    @Override
    public String toNetworkString()
    {
        ObjectMapper mapper = new ObjectMapper();
        String str = "";
        try
        {
            str = mapper.writeValueAsString(this);
        } catch (IOException e)
        {
            e.printStackTrace();
        }
        return str;
    }
}